appdirs.py

```diff
@@ -194,14 +212,19 @@ def user_config_dir(appname=None, appauthor=None, version=None, roaming=False):
     """
     if system == "win32":
         path = user_data_dir(appname, appauthor, None, roaming)
-    elif system == 'darwin':
-        path = os.path.expanduser('~/Library/Preferences/')
-        if appname:
-            path = os.path.join(path, appname)
     else:
-        path = os.getenv('XDG_CONFIG_HOME', os.path.expanduser("~/.config"))
-        if appname:
-            path = os.path.join(path, appname)
+        user = _effective_user()
```

Comment:
I think it's cleaner to have _effective_user() inside the format just like you did in user_state_dir() and user_log_dir()